### PR TITLE
My Site Dashboard: FAB + Quick Start tour auto-scroll

### LINKED ISSUE #17786: My Site Dashboard: Refine scrolling behavior for My Site
## Description 

Currently the My Site screen only scrolls on the child vc container. However, the entire screen should scroll.

## Tasks

- [x] Scroll the entire screen (Contain stackview in a scrollview, and may need to convert the BlogDetailsVC tableview into an IntrinsicTableView) #17820
- [x] Make sure the BlogDetailsVC scrolls to the correct element during the Quick Start tours #17847
- [x] Make sure the FAB appears on the My Site screen #17847

### PR DESCRIPTION
**Pull request by momo-ozawa**
Fixes #17786

## Description
- Adds the FAB to the My Site screen, if MSD is enabled (previously stuck to the bottom of BlogDetailsVC)
- Refactors the scrollToElement method in BlogDetailsVC to scroll to the correct Quick Start element, if MSD is enabled

## Notes
- These changes are behind the My Site Dashboard feature flag
- If you switch to the dashboard, the FAB will not do anything when you tap it

### Quick Start auto-scroll

1. Clean install
2. Log in
3. Tap "Show me around" in the post-epilogue Quick Start prompt
4. Make sure the MSD feature flag is enabled
5. Close then reopen the app
6. Go through every single Quick Start element for both check lists
7. ✅ Make sure the screen scrolls to the correct position for Quick Start elements in the site menu

https://user-images.githubusercontent.com/6711616/151860509-f90055f5-d002-4e5f-bc75-87b1c0189a4a.mp4

### FAB

1. Make sure the MSD feature flag is enabled
2. Go to My Site > Site menu
3. Tap on the FAB > Blog post
4. Create a draft post
5. ✅ Make sure that the draft post is there when you go to Posts > Drafts
6. Tap on the FAB > Site page
7. Create a draft page
8. ✅ Make sure the the draft page is there when you go to Pages > Drafts

https://user-images.githubusercontent.com/6711616/151860464-9c68e463-730a-44d3-b9d4-10f978870a5f.mp4

## Regression Notes
1. Potential unintended areas of impact
- Quick Start tour auto-scroll

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Made sure to wrap changes behind feature flags
- Tested the steps above 

3. What automated tests I added (or what prevented me from doing so)
- N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift [hunk 1]
```diff
@@ -193,16 +193,13 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
         }
 
         view.addSubview(scrollView)
+        view.pinSubviewToAllEdges(scrollView)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:197]: Turns out we need to pin the scroll view to the superview, NOT the safe area, in order for it to play nicely with all the inset logic in BlogDetailsVC

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift [hunk 2]
```diff
@@ -16,18 +16,25 @@ extension BlogDetailsViewController {
...
-                self?.scroll(to: element)
-            }
 
             if let info = notification.userInfo?[QuickStartTourGuide.notificationElementKey] as? QuickStartTourElement {
                 switch info {
                 case .noSuchElement:
-                    self?.additionalSafeAreaInsets = UIEdgeInsets.zero
+                    if FeatureFlag.mySiteDashboard.enabled,
+                       let parentVC = self?.parent as? MySiteViewController {
+                        parentVC.additionalSafeAreaInsets = UIEdgeInsets.zero
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift]:
```suggestion
                        parentVC.additionalSafeAreaInsets = .zero
```

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift [hunk 3]
```diff
@@ -16,18 +16,25 @@ extension BlogDetailsViewController {
...
+                    } else {
+                        self?.additionalSafeAreaInsets = UIEdgeInsets.zero
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift]:
```suggestion
                        self?.additionalSafeAreaInsets = .zero
```

**Review by leandroalonso — COMMENTED** [review]
@momo-ozawa it seems that there's an issue with the "View your site" (probably related to scrolling to the bottom?), see:

https://user-images.githubusercontent.com/7040243/152036107-34b59331-eb01-41f2-a9dd-233ce73f6eef.mp4

**Comment by momo-ozawa** [comment]
@leandroalonso Hmmmmm. I can't reproduce the View Site not scrolling. 

Here's what I did:
1. Log out
2. Log in
3. Show me around on Quick Start prompt
4. Select the View Site item when the checklist appears
5. Scrolls to View Site

https://user-images.githubusercontent.com/6711616/152130420-f44cffce-8b55-4aa5-851b-094a2f159a68.mp4

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift (hunk 3 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift]: Fixed! e644f7e

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+FancyAlerts.swift]: Fixed! e644f7e

**Comment by momo-ozawa** [comment]
@leandroalonso From your video, it looks like the MSD feature flag isn't enabled. There's actually a pre-existing issue for this: https://github.com/wordpress-mobile/WordPress-iOS/issues/16402

I think this PR should fix the View Site autoscroll issue!

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17847-e644f7e</code>. IPA is available <a href='https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/8de9ce56-ee0e-45bc-bd11-8bcbdf6f76bb/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=RfgY018bA3a9Zb%2F2Wnq25nzOeVv8n83ttbLJdwdlaxk%3D&se=2022-02-03T10%3A08%3A09Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17847-e644f7e</code>. IPA is available <a href='https://appcenter-filemanagement-distrib5ede6f06e.azureedge.net/962b56cb-513c-46a0-8821-abf7923071d1/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=YPlmdP2zFbMgy2IW3lpxbJjA%2FPHkFS1s6ch2sR8kKVA%3D&se=2022-02-03T10%3A11%3A05Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Review by leandroalonso — APPROVED** [review]
Ops! I assumed those changes would affect the current Quick Start.

Since that's not the case, :shipit:

**Pull request merged**